Google.Api.Generator/Generation/SnippetCodeGenerator.cs

```diff
@@ -184,55 +184,69 @@ private class SnippetDef
             // TODO: Reconsider this ugly hack when refactoring after removing the service-per-file snippets.
             private bool Canonical => RegionTagDisambiguation is null;
             private IEnumerable<ClientMethod.Types.Parameter> ClientMethodParameters { get; }
+            // May be null, in which case this snippet is for a single client method only.
+            private IEnumerable<ClientMethod.Types.Parameter> AdditionalClientMethodParameters { get; }
             private Func<SourceFileContext, bool, MethodDeclarationSyntax> MethodGenerator { get; }
 
             public SnippetDef(
-                MethodDetails targetMethod, bool sync, string snippetMethodName, IEnumerable<ClientMethod.Types.Parameter> clientMethodParameters,
+                MethodDetails targetMethod, bool sync, string snippetMethodName,
+                IEnumerable<ClientMethod.Types.Parameter> clientMethodParameters, IEnumerable<ClientMethod.Types.Parameter> additionalClientMethodParameters,
                 string regionTagDisambiguation, Func<SourceFileContext, bool, MethodDeclarationSyntax> methodGenerator) =>
-                (TargetMethod, Sync, SnippetMethodName, ClientMethodParameters, RegionTagDisambiguation, MethodGenerator) =
-                (targetMethod, sync, snippetMethodName, clientMethodParameters, regionTagDisambiguation, methodGenerator);
+                (TargetMethod, Sync, SnippetMethodName, ClientMethodParameters, AdditionalClientMethodParameters, RegionTagDisambiguation, MethodGenerator) =
+                (targetMethod, sync, snippetMethodName, clientMethodParameters, additionalClientMethodParameters, regionTagDisambiguation, methodGenerator);
 
             public MethodDeclarationSyntax GenerateMethod(SourceFileContext ctx) =>
                 // Include doc markers.
                 MethodGenerator(ctx, true)
                     .WithXmlDoc(XmlDoc.Summary($"Snippet for {TargetMethodName}"));
 
-            private SnippetMetadata GenerateMetadata() =>
-                new SnippetMetadata
+            private IEnumerable<SnippetMetadata> GenerateMetadata()
+            {
+                var metadata = GenerateMetadata(ClientMethodParameters);
+                if (AdditionalClientMethodParameters is null)
                 {
-                    Title = SnippetMethodName,
-                    Description = $"Snippet for {TargetMethodName}",
-                    Language = Language.CSharp,
-                    Canonical = Canonical,
-                    Origin = SnippetMetadata.Types.Origin.ApiDefinition,
-                    ClientMethod = new ClientMethod
+                    return new[] { metadata };
+                }
+
+                return new[] { metadata, GenerateMetadata(AdditionalClientMethodParameters) };
+
+                SnippetMetadata GenerateMetadata(IEnumerable<ClientMethod.Types.Parameter> clientMethodParameters) =>
```

Comment:
I'm assuming this large-looking change really isn't large at all, it's just a refactoring to a local method.